vslib/src/sai_vs_init.cpp

```diff
@@ -130,135 +127,349 @@ sai_status_t initialize_default_objects()
 
-            status = vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr);
+        attr.id = SAI_PORT_ATTR_HW_LANE_LIST;
+        attr.value.u32list.count = 4;
+        attr.value.u32list.list = &lanes[4 * i];
 
-            CHECK_STATUS(status);
+        CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
 
-            // populate lanes for each port
+        attr.id = SAI_PORT_ATTR_TYPE;
+        attr.value.s32 = SAI_PORT_TYPE_LOGICAL;
 
-            attr.id = SAI_PORT_ATTR_HW_LANE_LIST;
-            attr.value.u32list.count = 4;
-            attr.value.u32list.list = &lanes[4 * i];
+        CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
 
-            status = vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr);
+        // TODO populate other port attributes
+    }
 
-            CHECK_STATUS(status);
+    return SAI_STATUS_SUCCESS;
+}
 
-            // TODO populate ports attributes
-            // TODO populate vlan members
-        }
-    }
+sai_status_t create_port_list(std::vector<sai_object_id_t>& port_list)
+{
+    SWSS_LOG_ENTER();
 
     SWSS_LOG_INFO("create port list");
 
-    {
-        sai_attribute_t attr;
+    // TODO this is static, when we start to "create/remove" ports
+    // we need to update this list since it's dynamic
 
-        attr.id = SAI_SWITCH_ATTR_PORT_LIST;
-        attr.value.objlist.count = port_count;
-        attr.value.objlist.list = port_list.data();
+    sai_attribute_t attr;
 
-        status = vs_generic_set_switch(&attr);
+    uint32_t port_count = (uint32_t)port_list.size();
 
-        CHECK_STATUS(status);
+    attr.id = SAI_SWITCH_ATTR_PORT_LIST;
+    attr.value.objlist.count = port_count;
+    attr.value.objlist.list = port_list.data();
 
-        attr.id = SAI_SWITCH_ATTR_PORT_NUMBER;
-        attr.value.u32 = port_count;
+    CHECK_STATUS(vs_generic_set_switch(&attr));
 
-        status = vs_generic_set_switch(&attr);
+    attr.id = SAI_SWITCH_ATTR_PORT_NUMBER;
+    attr.value.u32 = port_count;
 
-        CHECK_STATUS(status);
-    }
+    return vs_generic_set_switch(&attr);
+}
+
+sai_status_t create_default_virtual_router()
+{
+    SWSS_LOG_ENTER();
 
     SWSS_LOG_INFO("create default virtual router");
 
+    sai_object_id_t virtual_router_id;
+
+    CHECK_STATUS(vs_generic_create(SAI_OBJECT_TYPE_VIRTUAL_ROUTER, &virtual_router_id, 0, NULL));
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID;
+    attr.value.oid = virtual_router_id;
+
+    return vs_generic_set_switch(&attr);
+}
+
+sai_status_t create_vlan_members_for_default_vlan(
+        std::vector<sai_object_id_t>& port_list,
+        std::vector<sai_object_id_t>& vlan_member_list)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("create vlan members for all ports");
+
+    vlan_member_list.clear();
+
+    vlan_members_map[DEFAULT_VLAN_NUMBER] = {};
+
+    for (auto &port_id : port_list)
```

Comment:
i doubt if we will add this to vs behavior. default vlan 1 needs to be there, but we could have 0 members in the default vlan. the behavior of adding all ports into default vlan by default is optional.